lib/uplinkc/object.go

```diff
@@ -278,6 +278,31 @@ func download(bucketRef C.BucketRef, path *C.char, cErr **C.char) C.DownloaderRe
 	})}
 }
 
+//export download_range
+// download_range returns an Object's data from specified range
+func download_range(bucketRef C.BucketRef, path *C.char, start, limit int64, cErr **C.char) C.DownloaderRef {
```

Comment:
would doing `path C.GoString` save you the call to `C.GoString()` on line 292, does that even work?

Reply:
path can be just `string` and that will be exported as `GoString` but I would prefer to stay consistent with rest of methods which are using `*C.char`

Reply:
:+1: perhaps we should make a ticket for a PR that simplifies the types.

Reply:
we released this library and we need to be backward compatible, might be not so easy.

Reply:
Just adding my two Cents: changing the types would mean to Regenerate my SWIG-wrapper, what I do all the time anything changes - so from my perspective this would not be a big deal.